x-pack/plugins/maps/public/index_pattern_util.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { indexPatternService } from './kibana_services';
+import { DECIMAL_DEGREES_PRECISION } from '../common/constants';
+import _ from 'lodash';
+
+
+export async function getIndexPatternsFromIds(indexPatternIds) {
+
+  const promises = [];
+  indexPatternIds.forEach((id) => {
+    const indexPatternPromise = indexPatternService.get(id);
+    if (indexPatternPromise) {
+      promises.push(indexPatternPromise);
+    }
+  });
+
+  return await Promise.all(promises);
+
+}
+
+
+export function createShapeFilter(geojsonPolygon, indexPatternId, geoField) {
+
+  //take outer
+  const points  = geojsonPolygon.coordinates[0].map(coordinatePair => {
+    return {
+      lon: _.round(coordinatePair[0], DECIMAL_DEGREES_PRECISION),
+      lat: _.round(coordinatePair[1], DECIMAL_DEGREES_PRECISION)
+    };
+  });
+  const field = geoField;
+  const filter = { meta: { negate: false, index: indexPatternId } };
+  filter.geo_polygon = { ignore_unmapped: true };
```

Comment:
How will this work with geo_shape indexes? `geo_polygon` only works for `geo_point`. Will need to craft [geo_shape](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-geo-shape-query.html) query for geo_shape indices

Reply:
Check out [createExtentFilter](https://github.com/elastic/kibana/blob/master/x-pack/plugins/maps/public/elasticsearch_geo_utils.js#L180) for an example of how we craft queries for geo_point/geo_shape for the extent filter